compiler/rustc_target/src/spec/base/lynxos_178.rs

```diff
@@ -0,0 +1,152 @@
+use std::borrow::Cow;
+use std::collections::BTreeMap;
+use std::env;
+use std::path::{Path, PathBuf};
+
+use crate::spec::{
+    Cc, CheckEnvironment, LinkerFlavor, Lld, PanicStrategy, RelocModel, RelroLevel, SplitDebuginfo,
+    StackProbeType, TargetOptions, cvs,
+};
+
+/// Strip mutability from &mut str.
+fn immutable(value: &mut str) -> &str {
+    let immutable = value;
+    immutable
+}
+
+macro_rules! cow_format {
+    ($arg:expr) => {
+        // Cow::from works with &str but not with &mut str.
+        Cow::from(immutable(Box::leak(format!($arg).into_boxed_str())))
+    };
+}
+
+struct EnvVar<'a> {
+    name: &'a str,
+    description: &'a str,
+}
+
+static ENV_PREFIX_VAR: EnvVar<'_> =
+    EnvVar { name: "ENV_PREFIX", description: "your LynxOS-178 Environment directory" };
+static GCC_DIR_VAR: EnvVar<'_> =
+    EnvVar { name: "GCC_DIR", description: "your LynxOS-178 GCC compiler directory" };
+
+struct LynxosPath<'a> {
+    env_var: &'a EnvVar<'a>,
+    rel_path: &'a str,
+}
+
+impl LynxosPath<'_> {
+    fn path(&self) -> Result<PathBuf, String> {
+        let var_value = env::var(self.env_var.name).map_err(|_| {
+            format!("Set {} to point at {}.", self.env_var.name, self.env_var.description)
+        })?;
+        Ok(Path::new(var_value.as_str()).join(self.rel_path))
+    }
+}
+
+// Paths relative to ENV_PREFIX
+static LINK_DIRS: [LynxosPath<'static>; 3] = [
+    LynxosPath { env_var: &GCC_DIR_VAR, rel_path: "" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "usr/lib/" },
+];
+static LINK_LIBS: [LynxosPath<'static>; 5] = [
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/crt1.o" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/crti.o" },
+    LynxosPath { env_var: &GCC_DIR_VAR, rel_path: "crtbegin.o" },
+    LynxosPath { env_var: &GCC_DIR_VAR, rel_path: "crtend.o" },
+    LynxosPath { env_var: &ENV_PREFIX_VAR, rel_path: "lib/crtn.o" },
+];
+
+pub(crate) fn check_environment() -> Result<(), String> {
```

Comment:
We don't want such eager checking. There's no reason why `rustc --target $target --emit obj` would need these checks.

I'd prefer to not have any advance checks and just let linking fail if something is wrong. While checks are friendlier for the user, they make the code more complicated and are, as mentioned above, prone to firing unnecessarily.